src/test_trie.py

```diff
@@ -0,0 +1,248 @@
+    init_trie_three_words_shared_letters.remove('be')
+    assert init_trie_three_words_shared_letters.contains('be') is False
+    init_trie_three_words_shared_letters.remove('bell')
+    assert init_trie_three_words_shared_letters.contains('bell') is False
+    assert init_trie_three_words_shared_letters.contains('bee') is True
+
+
+def test_size_trie_before_and_after_remove(init_trie_three_words_shared_letters):
+    """Test trie size before and after a removal."""
+    assert init_trie_three_words_shared_letters.size() == 3
+    init_trie_three_words_shared_letters.remove('bee')
+    assert init_trie_three_words_shared_letters.size() == 2
+
+
+def test_size_trie_before_and_after_insert(init_trie_three_words_shared_letters):
+    """Test trie size before and after a removal."""
+    assert init_trie_three_words_shared_letters.size() == 3
+    init_trie_three_words_shared_letters.insert('happy')
+    assert init_trie_three_words_shared_letters.size() == 4
+
+
+def test_insert_jumbo_list(init_jumbo_list):
+    """Test insert on jumbo word list."""
+    jumbo_trie = Trie()
+    for word in init_jumbo_list:
+        jumbo_trie.insert(word)
+    assert len(init_jumbo_list) == jumbo_trie.size()
+    assert jumbo_trie.contains('apple')
+
+
+def test_traversal_start_not_string(init_trie_three_words_shared_letters):
+    """Test traversal error when start arg is not a string. NOTE: currently returning a generator object."""
+    gen = init_trie_three_words_shared_letters.traversal(3)
+    with pytest.raises(TypeError):
+        next(gen)
+
+
+def test_traversal_start_empty_string(init_trie_three_words_shared_letters):
+    """Test traversal error when start arg is not a string. NOTE: currently returning a generator object."""
+    gen = init_trie_three_words_shared_letters.traversal('')
+    with pytest.raises(ValueError):
+        next(gen)
+
+
+def test_traversal_start_string_with_space(init_trie_three_words_shared_letters):
+    """Test traversal error when start arg is not a string. NOTE: currently returning a generator object."""
+    gen = init_trie_three_words_shared_letters.traversal(' ')
+    with pytest.raises(ValueError):
+        next(gen)
+
+
+def test_traversal_all_words(init_trie_three_words_shared_letters):
+    """Test all words with a shared start are generated."""
+    gen = init_trie_three_words_shared_letters.traversal('be')
+    assert next(gen) == 'be'
+    assert next(gen) == 'bee'
+    assert next(gen) == 'bell'
+
+
+def test_traversal_all_words_with_insert_not_in_gen(init_trie_three_words_shared_letters):
+    """Test all words with a shared start are generated."""
+    init_trie_three_words_shared_letters.insert('bacon')
+    gen = init_trie_three_words_shared_letters.traversal('be')
+    assert next(gen) == 'be'
+    assert next(gen) == 'bee'
+    assert next(gen) == 'bell'
+
+
+def test_traversal_all_words_with_insert_in_gen(init_trie_three_words_shared_letters):
```

Comment:
Don't you want to test that the word you inserted is in the output of your generator in this case? I woud also test the traversal by:
```
gen = [ x for x in init_trie_three_words_shared_letters.traversal('b')]
assert gen == ['be', 'bee', 'bell', 'bacon']
```